pkg/kube/finalizers.go

```diff
@@ -44,12 +46,17 @@ func RemoveFinalizer(ctx context.Context, obj client.Object, cl client.Client) e
 
 	err := cl.Update(ctx, obj)
 	if errors.IsNotFound(err) {
-		// We're reconciling a stale instance. The object no longer exists, so we're done.
-		return nil
+		log.Info("Resource no longer exists; nothing to do")
```

Comment:
I just wonder if these logs should be reduced to debug, if they are too verbose...

Reply:
But maybe we could remove the "Removing finalizer" statement and just leave the others (e.g. "Couldn't remove due to error", "Conflict", "Nothing to do", and "Finalizer removed").